code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -213,81 +218,84 @@
 
 
 //Attach a limb to a human and drop any existing limb of that type.
-/obj/item/bodypart/proc/replace_limb(mob/living/carbon/human/H, special)
-	if(!istype(H))
+/obj/item/bodypart/proc/replace_limb(mob/living/carbon/C, special)
+	if(!istype(C))
 		return
-	var/obj/item/bodypart/O = locate(src.type) in H.bodyparts
-
+	var/obj/item/bodypart/O = locate(src.type) in C.bodyparts
 	if(O)
 		O.drop_limb(1)
-	attach_limb(H, special)
+	attach_limb(C, special)
 
-/obj/item/bodypart/head/replace_limb(mob/living/carbon/human/H, special)
-	if(!istype(H))
+/obj/item/bodypart/head/replace_limb(mob/living/carbon/C, special)
+	if(!istype(C))
 		return
-	var/obj/item/bodypart/head/O = locate(src.type) in H.bodyparts
+	var/obj/item/bodypart/head/O = locate(src.type) in C.bodyparts
 	if(O)
 		if(!special)
 			return
 		else
 			O.drop_limb(1)
-	attach_limb(H, special)
+	attach_limb(C, special)
 
-/obj/item/bodypart/proc/attach_limb(mob/living/carbon/human/H, special)
+/obj/item/bodypart/proc/attach_limb(mob/living/carbon/C, special)
 	loc = null
-	owner = H
-	H.bodyparts += src
+	owner = C
+	C.bodyparts += src
 
 	if(special) //non conventional limb attachment
-		for(var/X in H.surgeries) //if we had an ongoing surgery to attach a new limb, we stop it.
+		for(var/X in C.surgeries) //if we had an ongoing surgery to attach a new limb, we stop it.
 			var/datum/surgery/S = X
 			var/surgery_zone = check_zone(S.location)
 			if(surgery_zone == body_zone)
-				H.surgeries -= S
+				C.surgeries -= S
 				qdel(S)
 				break
 
 	update_bodypart_damage_state()
-	H.updatehealth()
-	H.update_body()
-	H.update_hair()
-	H.update_damage_overlays()
-	H.update_canmove()
+	C.updatehealth()
+	C.update_body()
+	C.update_hair()
+	C.update_damage_overlays()
+	C.update_canmove()
 
 
-/obj/item/bodypart/r_arm/attach_limb(mob/living/carbon/human/H, special)
+/obj/item/bodypart/r_arm/attach_limb(mob/living/carbon/C, special)
 	..()
-	if(H.hud_used)
-		var/obj/screen/inventory/R = H.hud_used.inv_slots[slot_r_hand]
+	if(C.hud_used)
+		var/obj/screen/inventory/R = C.hud_used.inv_slots[slot_r_hand]
 		if(R)
 			R.update_icon()
+	C.update_inv_gloves()
```

Comment:
if we have only one arm, and we have have bloody hands, getting a second arm need to update the bloody hand overlay so both hands are bloody. (No we can't have one hand clean and one hand bloody, update_inv_gloves() would get called eventually)